### PR TITLE
AP_Common: Added termios header missing file.

### PR DESCRIPTION
**Pull request by hiro2233**
Added missing function cfsetspeed in termios header for C++Builder with Android NDK.

### CODE REVIEW

File: libraries/AP_Common/missing/termios.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include_next <termios.h>
+
+#if defined(__ANDROID__) && defined(__BORLANDC__)
+
+static __inline__ int cfsetspeed(struct termios *s, speed_t  speed)
```
Comment (lucasdemarchi) [inline libraries/AP_Common/missing/termios.h]: and you don't need to add underscores to inline

Comment (hiro2233) [inline libraries/AP_Common/missing/termios.h]: I tried, but nothing there, cfsetispeed() and cfsetospeed() are defined directly on API Level 19.

https://github.com/UrusTeam/android_ndk_toolchain_cross/blob/level_19/sysroot/usr/include/termios.h#L85

**Comment by hiro2233** [comment]
I reworked this one, please see my comments.

File: libraries/AP_Common/missing/termios.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_Common/missing/termios.h]: looks like they are defined in your link, so why can't we use them?

[the remark this answers is not in the document]
Comment (hiro2233) [reply libraries/AP_Common/missing/termios.h]: yes, this was a mistake on Embarcadero CBuilder when linking. This mistake was fixed on Urus Studio build system to work correctly with Firemonkey on Embarcadero Cbuilder.

Please doesn't merge it meanwhile, i have a better solution for this one, i will push it until URUS SHAL already working at 100% on Firemonkey.

File: libraries/AP_Common/missing/termios.h [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include_next <termios.h>
+
+#if defined(__ANDROID__) && !defined(HAVE_TERMIOS_CFSETSPEED)
+
```
Comment (lucasdemarchi) [inline libraries/AP_Common/missing/termios.h]: The check for `__ANDROID__` should not be needed...  `HAVE_TERMIOS_CFSETSPEED` will already tell if we need to implement.this missing function.

Comment (hiro2233) [reply libraries/AP_Common/missing/termios.h]: yes, this was added to compile without using waf, for that reason i put the double check. On build system like embarcadero is more difficult to build with waf internally, on Urus Studio this is not a problem.

Please doesn't merge it meanwhile, i have a better solution for this one, i will push it until URUS SHAL already working at 100% on Firemonkey.

Comment (lucasdemarchi) [reply libraries/AP_Common/missing/termios.h]: In the other build system you could simply let it undefined (hardcoded).  Here we only support waf

Comment (hiro2233) [reply libraries/AP_Common/missing/termios.h]: yes, i know that, __ANDROID__ was removed indeed, i will push the fix.

**Comment by khancyr** [comment]
Closed for the same reason as https://github.com/ArduPilot/ardupilot/pull/5393

**Pull request closed without merging**